spec/finish_unfinished_persons_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe FinishUnfinishedPersons do
+  describe '.compute_semi_id' do
+    let(:competition_year) { 2023 }
+    let(:available_per_semi) { {} }
+
+    before do
+      allow(Person).to receive_messages(
+        where: Person,
+        order: Person,
+        pick: nil,
+      )
```

Comment:
This is not how mocking during tests should work. What exactly is breaking if you remove this `before` block?